### ISSUE TITLE
Update to be compliant with Section 508 of the Americans with Disabilities Act

### ISSUE THREAD

**Issue by NASA-Michael** [Megabit/Blazorise#4686]
All US Government applications must be compliant with Section 508 of the ADA https://www.section508.gov/ it would be great if Blazorise was updated to be compliant (we know that it most likely won't be but we wanted to ask anyway)

**Comment by stsrki** [comment]
Blazorise is closely following Bootstrap methodology. While we don't support all accessibility rules out-of-the-box, you can still do them with Blazorise utility helpers. If you look at Bootstrap documentation for notes on accessibility, we can see what can be done https://getbootstrap.com/docs/5.0/getting-started/accessibility/.

### Structural markup

This is done mainly by the authors of the application. So we cannot have control over this.

### Interactive components

Modals, dropdowns, and other dynamic components should be working with all touch devices. But in case they are not we can always fix them. To be perfectly honest, I'm not an expert on accessibility.

### Color contrast

All our colors can be adjusted through theming. And also, each component can be adjusted with the `Background` and `TextColor` utilities.

### Visually hidden content

We have a `Visibility` parameter that can be used for this.

### Reduced motion

Things that are animated can also be disabled. For example, on `Modal`, you can set the animation timeout to zero.

**Comment by NASA-Michael** [comment]
Thank you very much for the reply. We will take a look at your suggestions

Michael Knopf
Software Architect
KSC Launchpad Technical POC
Cell: 352.262.2680
***@***.******@***.***>

From: Mladen Macanović ***@***.***>
Sent: Wednesday, March 22, 2023 5:21 AM
To: Megabit/Blazorise ***@***.***>
Cc: Knopf, Michael L (KSC-KIAC-3200)[ASRC Federal Data Solutions, LLC] ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [Megabit/Blazorise] Update to be compliant with Section 508 of the Americans with Disabilities Act (Issue #4686)


Blazorise is closely following Bootstrap methodology. While we don't support all accessibility rules out-of-the-box, you can still do them with Blazorise utility helpers. If you look at Bootstrap documentation for notes on accessibility, we can see what can be done https://getbootstrap.com/docs/5.0/getting-started/accessibility/<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgetbootstrap.com%2Fdocs%2F5.0%2Fgetting-started%2Faccessibility%2F&data=05%7C01%7Cmichael.knopf%40nasa.gov%7C462ae62e687c4328bd4c08db2ab6d026%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C638150736917575028%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=mt6RgkGM1d9yDtfVxM4uepr7ZNhiQ9QSUNIxKYaz9Ho%3D&reserved=0>.

Structural markup

This is done mainly by the authors of the application. So we cannot have control over this.

Interactive components

Modals, dropdowns, and other dynamic components should be working with all touch devices. But in case they are not we can always fix them. To be perfectly honest, I'm not an expert on accessibility.

Color contrast

All our colors can be adjusted through theming. And also, each component can be adjusted with the Background and TextColor utilities.

Visually hidden content

We have a Visibility parameter that can be used for this.

Reduced motion

Things that are animated can also be disabled. For example, on Modal, you can set the animation timeout to zero.

-
